zkevm-circuits/src/witness/block.rs

```diff
@@ -288,63 +286,16 @@ impl Block {
     // This helper returns bytecodes's whether `code_hash` is belong to first bytecode circuit.
     // always return true when feature 'dual_bytecode' is disabled.
     pub(crate) fn is_first_sub_bytecode_circuit(&self, code_hash: &U256) -> bool {
-        // bytecode_map should cover the target 'code_hash',
-        // but for extcodecopy, the external_address can be non existed code hash.
-        // `unwrap` here is not safe.
-        if self.bytecode_map.is_none() {
-            // not config feature 'dual_bytecode' case.
-            true
-        } else {
-            let bytecode_map = self
-                .bytecode_map
-                .as_ref()
-                .expect("bytecode_map is not none when enable 'dual_bytecode' feature");
-            *bytecode_map.get(code_hash).unwrap_or(&true)
-        }
-    }
-
-    // Get two sets of bytecodes for two bytecode sub circuits when enable feature 'dual_bytecode'.
-    pub(crate) fn get_bytecodes_for_dual_sub_circuits(&self) -> (Vec<&Bytecode>, Vec<&Bytecode>) {
-        if self.bytecode_map.is_none() {
-            println!("error: bytecode_map is none ");
-        }
-        let (first_subcircuit_bytecodes, second_subcircuit_bytecodes) =
-            Self::split_bytecodes_for_dual_sub_circuits(
-                &self.bytecodes,
-                self.bytecode_map
-                    .as_ref()
-                    .expect("bytecode_map is not none when enable feature 'dual_bytecode'"),
-            );
-
-        (first_subcircuit_bytecodes, second_subcircuit_bytecodes)
-    }
-
-    // Split two sets of bytecodes for two bytecode sub circuits.
-    //#[cfg(feature = "dual_bytecode")]
-    pub(crate) fn split_bytecodes_for_dual_sub_circuits<'a>(
-        bytecodes: &'a BTreeMap<Word, Bytecode>,
-        bytecode_map: &BTreeMap<Word, bool>,
-    ) -> (Vec<&'a Bytecode>, Vec<&'a Bytecode>) {
-        let mut first_subcircuit_bytecodes = Vec::<&Bytecode>::new();
-        let mut second_subcircuit_bytecodes = Vec::<&Bytecode>::new();
-
-        let first_subcircuit_code_hashes: Vec<Word> = bytecode_map
-            .iter()
-            .filter_map(|item| if *item.1 { Some(*item.0) } else { None })
-            .collect();
-
-        let _ = bytecodes
-            .iter()
-            .map(|code| {
-                if first_subcircuit_code_hashes.contains(code.0) {
-                    first_subcircuit_bytecodes.push(code.1)
-                } else {
-                    second_subcircuit_bytecodes.push(code.1)
-                }
-            })
-            .collect::<Vec<_>>();
-
-        (first_subcircuit_bytecodes, second_subcircuit_bytecodes)
+        self.bytecodes.get(code_hash).map_or(true, |b| {
+            #[cfg(feature = "dual_bytecode")]
```

Comment:
this is also what i mean feature declare  statement.
#[cfg(feature = "dual_bytecode")] {
....
}
#[cfg(not(feature = "dual_bytecode"))]{
.....
}